dynamic/service_test.go

```diff
@@ -49,3 +51,60 @@ func ExampleCreateShareURL() {
 	fmt.Printf("QR code: %s", result.ShareURL)
 	// Output: QR code: https://code.yoti.com/CAEaJDQzNzllZDc0LTU0YjItNDkxMy04OTE4LTExYzM2ZDU2OTU3ZDAC
 }
+
+func TestCreateShareURL_Unsuccessful_503(t *testing.T) {
+	_, err := createShareUrlWithErrorResponse(503, "SERVICE UNAVAILABLE")
+
+	assert.ErrorContains(t, err, "503: unknown HTTP error: SERVICE UNAVAILABLE")
+
+	tempError, temporary := err.(interface {
+		Temporary() bool
+	})
+	assert.Check(t, temporary && tempError.Temporary())
+}
+
+func TestCreateShareURL_Unsuccessful_404(t *testing.T) {
+	_, err := createShareUrlWithErrorResponse(404, "NOT FOUND")
+
+	assert.ErrorContains(t, err, "404: Application was not found: NOT FOUND")
```

Comment:
Are we set on this error layout? Or is this a recommended layout? I think it reads slightly better if we have:
```suggestion
	assert.ErrorContains(t, err, "404: NOT FOUND - Application was not found")
```
This has:
- Status code and reason together, which I think makes more logical sense
- A `-` instead of a 2nd `;`, I'm not a huge fan of having two `:` in one message

Reply:
I'll change the `:` to `-`, but in this case `NOT FOUND` is the HTTP response body, which is always appended

Reply:
@echarrod are you happy for me to change this?

From:
```
{CODE}: {MESSAGE}: {BODY}
```
to:
```
{CODE}: {MESSAGE} - {BODY}
```

Reply:
@echarrod I've made this change in f788eed - I've also update the test strings to make it clearer they are the response body

Reply:
Ok, yep looks good now